packages/@lwc/engine/src/framework/local-registry.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { appendVM, removeVM, getAssociatedVMIfPresent, renderVM } from './vm';
+import { isUndefined } from '@lwc/shared';
+import { startGlobalMeasure, GlobalMeasurementPhase, endGlobalMeasure } from './performance-timing';
+
+const globalRegisteredNames: Set<string> = new Set();
```

Comment:
Why do we need a list of all the registered custom elements, while the global CE registry is already doing that for us? Is it because `CustomElementRegistry.prototype.get` is not performant enough? We should add a comment above explaining why.

Reply:
yeah, that's the main reason, crossing that bridge unnecessarily in favor of a set lookup.